app/controllers/authenticate_controller.rb

```diff
@@ -9,13 +9,14 @@ def oidc_authenticate_code_redirect
     # params. This will likely need to be done via the Handler.
     params.permit!
 
-    auth_token = Authentication::Handler::AuthenticationHandler.new(
+    auth_token, headers_h = Authentication::Handler::AuthenticationHandler.new(
       authenticator_type: params[:authenticator]
     ).call(
       parameters: params.to_hash.symbolize_keys,
       request_ip: request.ip
     )
```

Comment:
Reconsidering if this is the interface we want. Authn-OIDC V2 is our only auth scheme that needs to set a response header - this means each other auth scheme converted to use the `AuthenticationHandler` pattern will be returning an empty set of headers.

To avoid passing the `response` object everywhere, I experimented with passing a lambda as an optional parameter to the `call` method, so that while other auth schemes can do this:
```rb
auth_token = Authentication::Handler::AuthenticationHandler.new(
  authenticator_type: params[:authenticator]
).call(
  parameters: params.to_hash.symbolize_keys,
  request_ip: request.ip
)
```

...Authn-OIDC V2 could do this:
```rb
auth_token = Authentication::Handler::AuthenticationHandler.new(
  authenticator_type: params[:authenticator]
).call(
  parameters: params.to_hash.symbolize_keys,
  request_ip: request.ip,
  set_header: lambda { |key, value|
    response.set_header(key, value) unless key.blank? or value.blank?
  }
)
```

... and the `set_header` block could be used in the OIDC-V2-specific `Authentication::AuthnOidc::V2::Strategy` class.